pkg/imgpkg/bundle/images_refs.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package bundle
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	ctlimg "github.com/k14s/imgpkg/pkg/imgpkg/image"
+	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/cppforlife/go-cli-ui/ui.UI
+
+type ImageRef struct {
+	lockconfig.ImageRef
+	IsBundle *bool
+}
+
+func NewImageRef(imgRef lockconfig.ImageRef, isBundle bool) ImageRef {
+	return ImageRef{ImageRef: imgRef, IsBundle: &isBundle}
+}
+
+func (i ImageRef) DeepCopy() ImageRef {
+	return ImageRef{
+		ImageRef: i.ImageRef.DeepCopy(),
+		IsBundle: i.IsBundle,
+	}
+}
+func (i *ImageRef) updateRepo(relativeToRepo string) *ImageRef {
```

Comment:
i'm curious... are these new line comments 'nits' or do you see these as required changes for this PR?

Reply:
it's a nit since it's a small stylistic suggestion. whether it's required is a tough question. in this case it's very atypical to have bunch of functions bunched up together without a newline separator (i have not seen that done in many go code bases) and given our existing code base style i would want these to get fixed unless there is a really good reason not to. imho its good to do a pass to fix things like these before putting PR in the review state.

Reply:
I did take a pass through and did intentionally group the functions for the structs together. I can adjust to different code styles. I think if this is a coding style for the existing code base (which I'm learning about now) I'll see If I can incorporate that into our linter. (as part of a separate PR)

Reply:
i'll also add the new lines